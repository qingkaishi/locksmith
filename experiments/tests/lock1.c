#include <pthread.h>
#include <stdio.h>

int x;

pthread_mutex_t l;

int f() {
  pthread_mutex_lock(&l);
  x++;
  pthread_mutex_unlock(&l);
}

int main() {
  pthread_t t;
  pthread_mutex_init(&l, NULL);
  pthread_mutex_init(&l, NULL);
  while(1)
  pthread_create(&t, NULL, &f, NULL);
  return 1;
}
