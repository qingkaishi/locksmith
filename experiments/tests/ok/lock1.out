************************* STARTING *************************


lock1.c:19: Warning: subtyping incompatible types:
int
void
shared: &x:lock1.c:4

Warning: Possible data race: &x:lock1.c:4 is protected by non-linear or concrete lock(s):
  l:lock1.c:6
  non linear concrete l:lock1.c:16
  non linear concrete l:lock1.c:17
 references:
  dereference of &x:lock1.c:4 at lock1.c:10
    &x:lock1.c:4
  locks acquired:
    l:lock1.c:6
    non linear concrete l:lock1.c:16
    non linear concrete l:lock1.c:17
  in: FORK at lock1.c:19 
             

  dereference of &x:lock1.c:4 at lock1.c:10
    &x:lock1.c:4
  locks acquired:
    l:lock1.c:6
    non linear concrete l:lock1.c:16
    non linear concrete l:lock1.c:17
  in: main at lock1.c:14 
             

*************************** DONE ***************************
